diff --git a/debian/configure b/debian/configure
index 73bad89cd..075a7d0dc 100755
--- a/debian/configure
+++ b/debian/configure
@@ -100,32 +100,32 @@ if [ ! -z "$RTAI_DEV" ]; then
 fi
 # libxenomai-dev does encode the architectures, but we have to resort to awful
 # shell juju to transform it into a proper dependency list
-XENOMAI_DEV="$(apt-cache search -n libxenomai-dev | head -1 | awk '{print $1}')"
-if [ ! -z "$XENOMAI_DEV" ]; then
-    if [ `apt-cache showsrc libxenomai-dev 2>/dev/null | wc -l` -eq 0 ]; then
-        cat 1>&2 <<-EOF
-	Package libxenomai-dev exists, but information about its source package
-	is not available.  This most likely means that you do not have the
-	right deb-src lines in /etc/apt, or that you need to "apt-get update".
-	EOF
-	exit 99
-    fi
-    XENOMAI_ARCHITECTURE="$(apt-cache showsrc libxenomai-dev \
-        | grep ^Architecture: \
-        | head -1 \
-        | tr '[:space:]' '\n' \
-        | tail -n +2 \
-        | grep -vw all \
-        | tr '\n' ' ')"
-    XENOMAI_DEV="$(apt-cache showsrc libxenomai-dev \
-        | grep ^Architecture: \
-        | head -1 \
-        | tr '[:space:]' '\n' \
-        | tail -n +2 \
-        | grep -vw all \
-        | sed "s/.*/$XENOMAI_DEV [&],/" \
-        | tr '\n' ' ')"
-fi
+#XENOMAI_DEV="$(apt-cache search -n libxenomai-dev | head -1 | awk '{print $1}')"
+#if [ ! -z "$XENOMAI_DEV" ]; then
+#    if [ `apt-cache showsrc libxenomai-dev 2>/dev/null | wc -l` -eq 0 ]; then
+#        cat 1>&2 <<-EOF
+#	Package libxenomai-dev exists, but information about its source package
+#	is not available.  This most likely means that you do not have the
+#	right deb-src lines in /etc/apt, or that you need to "apt-get update".
+#	EOF
+#	exit 99
+#    fi
+#    XENOMAI_ARCHITECTURE="$(apt-cache showsrc libxenomai-dev \
+#        | grep ^Architecture: \
+#        | head -1 \
+#        | tr '[:space:]' '\n' \
+#        | tail -n +2 \
+#        | grep -vw all \
+#        | tr '\n' ' ')"
+#    XENOMAI_DEV="$(apt-cache showsrc libxenomai-dev \
+#        | grep ^Architecture: \
+#        | head -1 \
+#        | tr '[:space:]' '\n' \
+#        | tail -n +2 \
+#        | grep -vw all \
+#        | sed "s/.*/$XENOMAI_DEV [&],/" \
+#        | tr '\n' ' ')"
+#fi
 
 while test $# -ne 0; do
     case "$1" in
diff --git a/src/rtapi/uspace_xenomai.cc b/src/rtapi/uspace_xenomai.cc
index 4da14675d..cee9fba08 100644
--- a/src/rtapi/uspace_xenomai.cc
+++ b/src/rtapi/uspace_xenomai.cc
@@ -1,7 +1,15 @@
 #include "config.h"
 #include "rtapi.h"
 #include "rtapi_uspace.hh"
+#ifdef XENOMAI_2X
 #include <posix/pthread.h>
+#else // For xenomai 3.x
+#include <trank/posix/pthread.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+#endif
 #include <atomic>
 #ifdef HAVE_SYS_IO_H
 #include <sys/io.h>
